### PR TITLE
Upgrade various old versions of rand

### PR DESCRIPTION
**Pull request by jdm**
Old versions of rand rely on Windows APIs that are forbidden in UWP applications. These changes upgrade various dependencies that transitively depend on them.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes
- [x] Fixes #22767.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23560)

### CODE REVIEW

**Comment by jdm** [comment]
@emilio Note that I'm making some changes to hashglobe and selectors in this PR.

**Comment by jdm** [comment]
Note to self: bring in https://github.com/servo/servo/pull/22767/commits/420a8996810b8d212352a523edb115cb097dd82d before closing #22767.

**Comment by asajeffrey** [comment]
```
error: use of deprecated item 'uuid::Uuid::from_random_bytes': please use the `uuid::Builder` instead to set the variant and version
   --> components/rand/lib.rs:165:5
    |
165 |     Uuid::from_random_bytes(bytes)
    |     ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D deprecated` implied by `-D warnings`
```

**Comment by jdm** [comment]
Ready for review.

File: servo-tidy.toml [hunk 1]
```diff
@@ -32,15 +32,10 @@ rand = [
...
-  "base64",
-  "block-buffer",
-  "byte-tools",
   "crossbeam-deque",
   "crossbeam-epoch",
   "crossbeam-utils",
-  "digest",
   "dwrote",
-  "generic-array",
   "gl_generator", # https://github.com/servo/servo/pull/23288#issuecomment-494687746
```
Comment (asajeffrey) [inline servo-tidy.toml]: Yay fewer dupes!

**Review by asajeffrey — APPROVED** [review]

**Comment by asajeffrey** [comment]
What a lot of renaming and renumbering! @bors-servo r+

**Comment by CYBAI** [comment]
Failures in Travis and AppVeyor 👀

<details>

```rust
   Compiling net v0.0.1 (/home/travis/build/servo/servo/components/net)
error[E0432]: unresolved import `headers_core::HeaderMapExt`
 --> components/net/tests/data_loader.rs:6:5
  |
6 | use headers_core::HeaderMapExt;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HeaderMapExt` in the root
error[E0432]: unresolved import `headers_core::HeaderMapExt`
  --> components/net/tests/fetch.rs:15:5
   |
15 | use headers_core::HeaderMapExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HeaderMapExt` in the root
error[E0432]: unresolved import `headers_core::HeaderMapExt`
  --> components/net/tests/http_loader.rs:17:5
   |
17 | use headers_core::HeaderMapExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HeaderMapExt` in the root
error[E0432]: unresolved import `headers_ext`
 --> components/net/tests/data_loader.rs:7:5
  |
7 | use headers_ext::ContentType;
  |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error[E0432]: unresolved import `headers_ext`
  --> components/net/tests/fetch.rs:16:5
   |
16 | use headers_ext::{
   |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
   Compiling gfx v0.0.1 (/home/travis/build/servo/servo/components/gfx)
error[E0432]: unresolved import `headers_ext`
  --> components/net/tests/fetch.rs:19:5
   |
19 | use headers_ext::{AccessControlAllowMethods, AccessControlMaxAge};
   |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error[E0432]: unresolved import `headers_ext`
  --> components/net/tests/fetch.rs:20:5
   |
20 | use headers_ext::{
   |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error[E0432]: unresolved import `headers_ext`
  --> components/net/tests/http_loader.rs:18:5
   |
18 | use headers_ext::{
   |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error[E0432]: unresolved import `headers_ext`
  --> components/net/tests/http_loader.rs:21:5
   |
21 | use headers_ext::{StrictTransportSecurity, UserAgent};
   |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error: unused import: `std::iter::FromIterator`
  --> components/net/tests/fetch.rs:45:5
   |
45 | use std::iter::FromIterator;
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
error[E0599]: no method named `typed_get` found for type `http::header::map::HeaderMap` in the current scope
  --> components/net/tests/data_loader.rs:34:56
   |
34 |             let header_content_type = response.headers.typed_get::<ContentType>();
   |                                                        ^^^^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
           `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:189:10
    |
189 |         .typed_get::<ContentType>()
    |          ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:194:64
    |
194 |     let content_length: ContentLength = fetch_response.headers.typed_get().unwrap();
    |                                                                ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:217:10
    |
217 |         .typed_get::<ContentType>()
    |          ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:277:18
    |
277 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:280:18
    |
280 |                 .typed_insert(AccessControlAllowCredentials);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:283:18
    |
283 |                 .typed_insert(AccessControlAllowMethods::from_iter(vec![Method::GET]));
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:287:18
    |
287 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:328:18
    |
328 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:331:18
    |
331 |                 .typed_insert(AccessControlAllowCredentials);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:334:18
    |
334 |                 .typed_insert(AccessControlAllowMethods::from_iter(vec![Method::GET]));
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:337:18
    |
337 |                 .typed_insert(AccessControlMaxAge::from(Duration::new(6000, 0)));
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:341:18
    |
341 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:393:18
    |
393 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:396:18
    |
396 |                 .typed_insert(AccessControlAllowCredentials);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:399:18
    |
399 |                 .typed_insert(AccessControlAllowMethods::from_iter(vec![Method::GET]));
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:403:18
    |
403 |                 .typed_insert(AccessControlAllowOrigin::ANY);
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:462:14
    |
462 |             .typed_insert(AccessControlAllowOrigin::ANY);
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:465:32
    |
465 |         response.headers_mut().typed_insert(CacheControl::new());
    |                                ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:472:14
    |
472 |             .typed_insert(ContentType::from(mime::TEXT_HTML));
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:475:14
    |
475 |             .typed_insert(Expires::from(SystemTime::now() + Duration::new(86400, 0)));
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:478:14
    |
478 |             .typed_insert(LastModified::from(SystemTime::now()));
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:479:32
    |
479 |         response.headers_mut().typed_insert(Pragma::no_cache());
    |                                ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:491:14
    |
491 |             .typed_insert(AccessControlAllowHeaders::from_iter(vec![
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/fetch.rs:754:36
    |
754 |             response.headers_mut().typed_insert(mime_header);
    |                                    ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
    --> components/net/tests/fetch.rs:1079:13
     |
1079 |     headers.typed_insert(Host::from(
     |             ^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
    --> components/net/tests/fetch.rs:1092:13
     |
1092 |     headers.typed_insert::<UserAgent>(DEFAULT_USER_AGENT.parse().unwrap());
     |             ^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
    --> components/net/tests/fetch.rs:1109:22
     |
1109 |     response_headers.typed_insert(ContentLength(content.len() as u64));
     |                      ^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:115:13
    |
115 |     headers.typed_insert(Host::from(
    |             ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:133:13
    |
133 |     headers.typed_insert::<UserAgent>(crate::DEFAULT_USER_AGENT.parse().unwrap());
    |             ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:156:18
    |
156 |     post_headers.typed_insert(ContentLength(0 as u64));
    |                  ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `&http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:189:31
    |
189 |             request.headers().typed_get::<ContentLength>(),
    |                               ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:220:14
    |
220 |             .typed_insert(Host::from("foo.bar".parse::<Authority>().unwrap()));
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:226:21
    |
226 |     request_headers.typed_insert(Host::from("bar.foo".parse::<Authority>().unwrap()));
    |                     ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:259:13
    |
259 |     headers.typed_insert(Host::from(
    |             ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:277:13
    |
277 |     headers.typed_insert::<UserAgent>(crate::DEFAULT_USER_AGENT.parse().unwrap());
    |             ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:294:22
    |
294 |     response_headers.typed_insert(ContentLength(content.len() as u64));
    |                      ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:295:22
    |
295 |     response_headers.typed_insert(Host::from("foo.bar".parse::<Authority>().unwrap()));
    |                      ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:296:22
    |
296 |     response_headers.typed_insert(
    |                      ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `&http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:301:14
    |
301 |             .typed_get::<Date>()
    |              ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:322:14
    |
322 |             .typed_insert(Host::from("foo.bar".parse::<Authority>().unwrap()));
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:554:14
    |
554 |             .typed_insert(StrictTransportSecurity::excluding_subdomains(
    |              ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `&http::header::map::HeaderMap` in the current scope
   --> components/net/tests/http_loader.rs:819:31
    |
819 |             request.headers().typed_get::<ContentLength>(),
    |                               ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `&http::header::map::HeaderMap` in the current scope
    --> components/net/tests/http_loader.rs:1282:31
     |
1282 |             request.headers().typed_get::<Authorization<Basic>>(),
     |                               ^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_insert` found for type `&mut http::header::map::HeaderMap` in the current scope
    --> components/net/tests/http_loader.rs:1358:14
     |
1358 |             .typed_insert(AccessControlAllowOrigin::ANY);
     |              ^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error[E0599]: no method named `typed_get` found for type `&http::header::map::HeaderMap` in the current scope
    --> components/net/tests/http_loader.rs:1359:33
     |
1359 |         match request.headers().typed_get::<Origin>() {
     |                                 ^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
             `use headers::map_ext::HeaderMapExt;`
error: aborting due to 56 previous errors
Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: Could not compile `net`.
warning: build failed, waiting for other jobs to finish...
error: build failed
The command "./mach test-unit" exited with 101.
```

</details>

**Comment by CYBAI** [comment]
Travis complains about tidy 👀

```diff
Diff in /home/travis/build/servo/servo/components/net/tests/fetch.rs at line 12:
 use crossbeam_channel::{unbounded, Sender};
 use devtools_traits::HttpRequest as DevtoolsHttpRequest;
 use devtools_traits::HttpResponse as DevtoolsHttpResponse;
-use headers::{
-    AccessControlAllowCredentials, AccessControlAllowHeaders, AccessControlAllowOrigin,
-};
+use headers::{AccessControlAllowCredentials, AccessControlAllowHeaders, AccessControlAllowOrigin};
 use headers::{AccessControlAllowMethods, AccessControlMaxAge, HeaderMapExt};
 use headers::{
     CacheControl, ContentLength, ContentType, Expires, Host, LastModified, Pragma, UserAgent,
Run `./mach fmt` to fix the formatting
```

AppVeyor and macOS build in TC failed with

```rust
error[E0432]: unresolved import `headers_core::HeaderMapExt`
 --> components\net\tests\data_loader.rs:6:5
  |
6 | use headers_core::HeaderMapExt;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HeaderMapExt` in the root
error[E0432]: unresolved import `headers::Basic`
  --> components\net\tests\http_loader.rs:18:46
   |
18 |     AccessControlAllowOrigin, Authorization, Basic, ContentLength, Date, HeaderMapExt, Host,
   |                                              ^^^^^ no `Basic` in the root
error[E0432]: unresolved import `headers_ext`
 --> components\net\tests\data_loader.rs:7:5
  |
7 | use headers_ext::ContentType;
  |     ^^^^^^^^^^^ use of undeclared type or module `headers_ext`
error[E0599]: no method named `typed_get` found for type `http::header::map::HeaderMap` in the current scope
  --> components\net\tests\data_loader.rs:34:56
   |
34 |             let header_content_type = response.headers.typed_get::<ContentType>();
   |                                                        ^^^^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
           `use headers::map_ext::HeaderMapExt;`
error: aborting due to 4 previous errors
Some errors have detailed explanations: E0432, E0599.
```

**Comment by CYBAI** [comment]
Travis failed with this 👀

```
error[E0432]: unresolved import `headers::Basic`
  --> components/net/tests/http_loader.rs:18:46
   |
18 |     AccessControlAllowOrigin, Authorization, Basic, ContentLength, Date, HeaderMapExt, Host,
   |                                              ^^^^^ no `Basic` in the root
error: aborting due to previous error
For more information about this error, try `rustc --explain E0432`.
error: Could not compile `net`.
warning: build failed, waiting for other jobs to finish...
error: build failed
The command "./mach test-unit" exited with 101.
```

**Comment by jdm** [comment]
Ugh.

**Comment by jdm** [comment]
I'm just going to wait for travis this time.

**Pull request merged**